src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -0,0 +1,664 @@
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        [CLSCompliant(false)]
+        public void SetSByte(sbyte value) => _value = value.ToString();
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="ushort"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        [CLSCompliant(false)]
+        public void SetUInt16(ushort value) => _value = value.ToString();
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="uint"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        [CLSCompliant(false)]
+        public void SetUInt32(uint value) => _value = value.ToString();
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="ulong"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        [CLSCompliant(false)]
+        public void SetUInt64(ulong value) => _value = value.ToString();
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="decimal"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        public void SetDecimal(decimal value) => _value = value.ToString();
+
+        /// <summary>
+        ///   Converts a <see cref="byte"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(byte value) => new JsonNumber(value);
+
+        /// <summary>
+        ///   Converts a <see cref="short"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(short value) => new JsonNumber(value);
+
+        /// <summary>
+        ///   Converts an <see cref="int"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(int value) => new JsonNumber(value);
+
+        /// <summary>
+        ///   Converts a <see cref="long"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(long value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a <see cref="float"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(float value) => new JsonNumber(value);
```

Comment:
Hmm... this implicit cast can throw. Something worth thinking about since it goes against the FDG:
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/operator-overloads#conversion-operators

```C#
JsonNumber number = float.PositiveInfinity; // throws
```

cc @bartonjs